dev/NavigationView/NavigationViewItemsFactory.h

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#pragma once
+
+#include "common.h"
+#include "ElementFactory.h"
+
+class NavigationViewItemsFactory :
+    public winrt::implements<NavigationViewItemsFactory, ElementFactory>
+{
+public:
+    NavigationViewItemsFactory();
+
+    void UserElementFactory(winrt::Windows::UI::Xaml::IElementFactory elementFactory) { m_userElementFactory = elementFactory; };
+
+#pragma region IElementFactoryOverrides
+    winrt::UIElement GetElementCore(winrt::ElementFactoryGetArgs const& args);
+    void RecycleElementCore(winrt::ElementFactoryRecycleArgs const& args);
+#pragma endregion
+
+private:
+    winrt::Windows::UI::Xaml::IElementFactory m_userElementFactory{ nullptr };
```

Comment:
>m_userElementFactory [](start = 46, length = 20)

Does this need to be a tracker ref?

Reply:
still wondering if this needs to be a tracker ref

---
In reply to: [353374304](https://github.com/microsoft/microsoft-ui-xaml/pull/1683#discussion_r353374304) [](ancestors = 353374304)

Reply:
sorry, forgot to reply. At this point, I don't believe so. Primarily because I don't see how this would be a problem (for example, introducing a reference cycle).

If you look at the `ItemTemplateWrapper` object we instantiate in the method `UserElementFactory`, it holds a strong reference to the passed in datatemplate/datatemplateselector. So essentially, unless we update `ItemTemplateWrapper` to be able to use weak refs, whenever we hit this code path, we still retain a strong ref to the datatemplate. 

Also, as an fyi, here is an example in Repeater that I worked off of, hence the same scenario:
[https://github.com/microsoft/microsoft-ui-xaml/blob/32133a392e1a3c889f8285e83717ec19e35727b1/dev/Repeater/ItemsRepeater.h#L144](https://github.com/microsoft/microsoft-ui-xaml/blob/32133a392e1a3c889f8285e83717ec19e35727b1/dev/Repeater/ItemsRepeater.h#L144)

---
In reply to: [364880204](https://github.com/microsoft/microsoft-ui-xaml/pull/1683#discussion_r364880204) [](ancestors = 364880204,353374304)